app/code/Magento/Sales/Model/Order/Email/Sender.php

```diff
@@ -157,9 +166,20 @@ protected function getTemplateOptions()
      */
     protected function getFormattedShippingAddress($order)
     {
-        return $order->getIsVirtual()
+       // echo $order->getStore()->getId(); die("dss");
+        $writer = new \Zend\Log\Writer\Stream(BP . '/var/log/test11.log');
+        $logger = new \Zend\Log\Logger();
+        $logger->addWriter($writer);
+        $logger->info($order->getStore()->getStoreId());
+        $this->_appEmulation->startEnvironmentEmulation($order->getStoreId(),\Magento\Framework\App\Area::AREA_FRONTEND,1);
+        
+        $shippingAddress = $order->getIsVirtual()
             ? null
             : $this->addressRenderer->format($order->getShippingAddress(), 'html');
+
+        $this->_appEmulation->stopEnvironmentEmulation();
```

Comment:
1. Could you wrap it to try finally block in order to prevent not stopped emulation in case of exception? Something like this:
```suggestion
        try {
            $this->_appEmulation->startEnvironmentEmulation($order->getStoreId(),\Magento\Framework\App\Area::AREA_FRONTEND,1);

            $shippingAddress = $order->getIsVirtual()
                ? null
                : $this->addressRenderer->format($order->getShippingAddress(), 'html');
        } finally {
            $this->_appEmulation->stopEnvironmentEmulation();
        }
```

2. Could you explain why we need to have 3rd param 1? 
- first of all it should be true or false
- why do we need to pass "force" param here?

Reply:
yes the second parameter should to boolean . I had to add `force` parameter in `startEnvironmentEmulation`  function to make it work . without that the emulation was not working. 
I checked it more it seems the emulation is also being used in `Magento_Email` module so we have to use  "force" param here but I am not sure about it.

Reply:
It means that we’re trying to emulate area inside emulated area. Seems like this is not really expected behavior. Could you check it ?

Reply:
It means that we’re trying to start emulation inside emulation. Seems like this is not really expected behavior. Could you check it ?

Reply:
Yes I am checking more of it. Will check for another possible solution for this issue